### PR TITLE
refactor(meta): split meta into smaller crates (step 1)

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Just trying my idea quickly. Feel free to leave any comments/concerns. Will proceed if the idea sounds good.

I came up with this idea because I thought `async_trait` is a major reason of the slow compilation (https://github.com/risingwavelabs/risingwave/issues/9553#issuecomment-1766602371). There are many `async_trait`s in tonic services, and many heavy futures to type check. We can hardly get rid of them, but maybe we can split the services into crates so that we can compile them in parallel. (a little similar to #12485, but different)

As a first step, split out:
- `meta/service`: tonic grpc service implementations. We may further split this into parallel sub-crates.
- `meta/node`: the server binary

### Result
A rough manual test:
compiling only meta takes 50s -> 42s
compiling pb -> cmd_all takes 92s -> 76s

Before
image

After
image

### Note
In order to quickly pass compilation, some brute force is used:
- All `pub(crate)` are simply replaced with `pub`.
- `use risingwave_meta::*` is used without careful fine-graind imports.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/rpc/mod.rs [hunk 1]
```diff
@@ -12,19 +12,30 @@
...
 
 pub use election::etcd::EtcdElectionClient;
 pub use election::{ElectionClient, ElectionMember};
-pub use service::cluster_service::ClusterServiceImpl;
-pub use service::ddl_service::DdlServiceImpl;
-pub use service::heartbeat_service::HeartbeatServiceImpl;
-pub use service::hummock_service::HummockServiceImpl;
-pub use service::notification_service::NotificationServiceImpl;
-pub use service::stream_service::StreamServiceImpl;
+// pub use service::cluster_service::ClusterServiceImpl;
```
Comment (wenym1) [inline src/meta/src/rpc/mod.rs:36]: Remove the comment?

**Review by wenym1 — APPROVED** [review]
Rest LGTM.

File: src/meta/src/rpc/mod.rs (hunk 1 above)
Comment (xxchan) [reply src/meta/src/rpc/mod.rs:36]: Yes, I will take another look and refine such minor issues later if this idea is generally ok. 🥵

**Review by fuyufjh — APPROVED** [review]
LGTM

**Review by yezizp2012 — APPROVED** [review]
LGTM!

**Review by BugenZhao — COMMENTED** [review]
LGTM. How were you able to do this so quickly?

**Comment by xxchan** [comment]
> How were you able to do this so quickly?

🤔️ It took me about 2h to do this. Steps were:

1. Simply mv `meta/src/rpc/service`, to `meta_service` and `meta/src/lib.rs`, `meta/src/rpc/server.rs` to `meta_node`. 
    - `Cargo.toml` is copied from meta (without pruning the dependencies), and add meta as dependency.
    - Add `use risingwave_meta::*` to new crates so that we don't need to change `use crate::..`.
2. make `cargo build -p risingwave_meta` compile. IIRC this is very easy, thanks to the fact that the dependency relationship of these are very clear.
3. make `cargo build -p risingwave_meta_service` compile. Mainly just need to change `pub(crate)` to `pub`... Then fix a few minor issues. IIRC this part took the most time.
4. make `cargo build -p risingwave_meta_node` compile. Easier. Mainly just replace imports.
5. make `cargo build -p risingwave_cmd_all` compile.
6. `risedev check-fix`, and fix a few minor issues.

As you can see the diff of this PR is also very reviewable, so it's a relatively easy refactor. I think it should mainly because the dependency `node <- service <- others` is clear.

**Pull request merged**